src/main/java/org/opensearch/knn/plugin/transport/TrainingModelRequest.java

```diff
@@ -336,5 +355,10 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeInt(this.maximumVectorCount);
         out.writeInt(this.searchSize);
         out.writeInt(this.trainingDataSizeInKB);
+        if (IndexUtil.isVersionOnOrAfterMinRequiredVersion(out.getVersion(), KNNConstants.MODEL_VECTOR_DATA_TYPE_KEY)) {
+            out.writeString(this.vectorDataType.getValue());
+        } else {
+            out.writeString(VectorDataType.FLOAT.getValue());
```

Comment:
This is not going to work. The old node does not know how to parse this vector data type.
We should throw an error that binary data type is not supported when cluster has a node of old version.
Also, I think we also don't support Byte data type. Have we add a validation for that?

Reply:
Added version validation at rest request step.